### DISCUSSION TITLE
[2024-11-11] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#11524] [Weekly Demo Day]
* #11435 
* #11448

[Screencast From 2024-10-30 13-45-48.webm](https://github.com/user-attachments/assets/9ed2384a-de85-4059-a0fe-c0a45b2fe07c)

* #11492 - to close dependabot PR and reduce random GUI test failures.
* Investigated #11493 - the root cause is y-doc server version mismatch between newest nightlies (develop builds) and cloud ydoc-server. We plan to add version check to display at least a warning soon (after December release). How to properly handle version mismatch is still to be discussed.
* In progress: #10059 

https://github.com/user-attachments/assets/9c7366fb-cd34-4101-a3e5-01347b1345ad

**Comment by JaroslavTulach** [comment]
Engine news:
- Prevent **loss of component positions**: #11390
- Discuss: #11366
- `docker` image with **ydoc-server** & Graal.js: #11517
- Demo (see [steps](https://github.com/enso-org/enso/pull/11415#issuecomment-2453902666)): #11415

**Comment by vitvakatu** [comment]
- https://github.com/enso-org/enso/pull/11451 – CB displays methods of Any now, and the mechanism is general enough to work in other cases where we have `parentType` information available.
- https://github.com/enso-org/enso/pull/11494 – waits for merge because of unrelated issues on develop. We prevent visualizations from ever receiving invalid data (__invalid__ here means “stale data from another visualization type”). We are also aware of current issues with error handling inside custom visualizations (WIP) and found a bug in Vue that was mitigated.
- https://github.com/enso-org/enso/pull/11523 – rework (or rather a replacement) of the `Record` button on components.

**Comment by kazcw** [comment]
- #11302
  - Introduce expression/statement distinction
  - Simplifications of frontend AST APIs and implementation
- #11469
  - Improve AST type safety
  - Fix right dock size and animation bugs

**Comment by Frizi** [comment]
- https://github.com/enso-org/enso/issues/10966 (currently broken for many people :( fix TBD)
- Added empty state view when the graph disappeared - https://github.com/enso-org/enso/issues/11211
<img width="343" alt="image" src="https://github.com/user-attachments/assets/0bdb0cff-c24d-4238-ab9f-9a94bbf888ba">

- Added group opening to the node context menu - https://github.com/enso-org/enso/issues/11179
<img width="188" alt="image" src="https://github.com/user-attachments/assets/51901a73-e283-4be6-8776-6a3a9069ea8f">

- Working on function signature editor in documentation panel:
<img width="341" alt="image" src="https://github.com/user-attachments/assets/fbda183c-2b5a-4dce-a46a-2f19962eff6d">
